src/loader/bgw_launcher.c

```diff
@@ -591,6 +591,12 @@ launcher_handle_message(HTAB *db_htab)
 static void
 bgw_sigterm(SIGNAL_ARGS)
 {
+	/*
+	 * The launcher should never be in a critical section, but if it is we
+	 * have no choice but to PANIC since it is unsafe to exit.
+	 */
+	if (CritSectionCount != 0)
```

Comment:
Is it worth it to call into Postgres' `bgworker_die` here after our LOG message below, since I believe in this case `bgworker_die` will do the right thing since you'll want to panic in a critical section.

Reply:
I don't think that will work as bgworker_die does not proc_exit with code 0.